### PR TITLE
Fix for #configuration builtin

### PR DESCRIPTION
**Pull request by theo25**

### CODE REVIEW

**Comment by daejunpark** [comment]
@theo25 let's see if it passes the tests:
https://github.com/kframework/k/commits/theo-reflection-fix

**Comment by daejunpark** [comment]
@theo25 says:

> So, TermContext has a private field called topTerm that was never set and kept its default null value. The #configuration builtin attempts to access this field to get the current configuration but since it’s null, a NullPointerException is thrown and never gets caught. So I added the correct (I think) initialization of the field when a ConstrainedTerm is built out of a Term and a TermContext.
>
> Also, I have changed the implementation of the builtin. I am not sure what a KLabelInjection is but it did not return a term with a KLabel corresponding to the top cell of the configuration (e.g. “\<T\>”). Instead it was returning term with a KLabel that somehow was made out of the whole configuration (e.g. “#\<T\>(\<k\> ...)”). I replaced that with simply returning the Term stored in the topTerm field of the context.

**Comment by daejunpark** [comment]
The jenkins failure seems to be due to ng setup failure. @ehildenb could you confirm that we can ignore this failure to merge this?

**Comment by ehildenb** [comment]
I'm looking into it, will try to re-trigger the test with corrected settings.

**Comment by theo25** [comment]
@daejunpark FYI, I tried this fix on top of keq with our equivalence checking examples and everything works fine.

**Review by daejunpark — APPROVED** [review]

File: java-backend/src/main/java/org/kframework/backend/java/builtins/MetaK.java [hunk 1]
```diff
@@ -200,6 +200,7 @@ public static KItem getKLabel(KItem kItem, TermContext context) {
     }
 
     public static Term configuration(TermContext context) {
-        return KLabelInjection.injectionOf(context.getTopTerm(), context.global());
+        //return KLabelInjection.injectionOf(context.getTopTerm(), context.global());
+        return context.getTopTerm();
```
Comment (daejunpark) [inline java-backend/src/main/java/org/kframework/backend/java/builtins/MetaK.java:204]: `KLabelInjection` is somewhat legacy and kind of deprecated. It should be OK to drop.

File: java-backend/src/main/java/org/kframework/backend/java/kil/ConstrainedTerm.java [hunk 2]
```diff
@@ -81,6 +81,7 @@ public boolean equals(Object obj) {
     public ConstrainedTerm(Term term, ConjunctiveFormula constraint, TermContext context) {
         this.data = new Data(term, constraint);
         this.context = context;
+        this.context.setTopTerm(term);
```
Comment (daejunpark) [inline java-backend/src/main/java/org/kframework/backend/java/kil/ConstrainedTerm.java:84]: `TermContext.topTerm` is never used, so this change shouldn't affect others.

**Review by daejunpark — APPROVED** [review]

**Comment by daejunpark** [comment]
The jenkins failure has nothing to do with this PR and I merge this.

**Pull request merged**